File: scripts/check_policies.py
import os
import asyncio
import asyncpg
import urllib.parse

with open('.env', 'r') as f:
    for line in f:
        if line.startswith('DATABASE_URL='):
            db_url = line.strip().split('=', 1)[1]
            if db_url.startswith('"') and db_url.endswith('"'):
                db_url = db_url[1:-1]
            break

async def main():
    conn = await asyncpg.connect(db_url)
    rows = await conn.fetch("SELECT policyname, qual FROM pg_policies WHERE tablename IN ('courses', 'lectures');")
    for row in rows:
        print(f"{row['policyname']}: {row['qual']}")
    await conn.close()

if __name__ == '__main__':
    asyncio.run(main())

File: scripts/check_policies.ts
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
import { resolve } from 'path';

dotenv.config({ path: resolve(process.cwd(), '.env') });

const supabaseUrl = process.env.VITE_SUPABASE_URL!;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY!;
const supabase = createClient(supabaseUrl, supabaseKey);

async function run() {
  const { data, error } = await supabase.from('courses').select('*');
  console.log("COURSES:", data?.length);
  
  // let's fetch policies
  const { data: policies, error: err } = await supabase.rpc('execute_sql', { query: "SELECT * FROM pg_policies WHERE tablename = 'courses' OR tablename = 'lectures'" });
  if (err) {
    console.log("Could not use rpc execute_sql", err);
  } else {
    console.log(policies);
  }
}
run();
